source/generate_source_file.py

```diff
@@ -0,0 +1,209 @@
+                          '飞', '艺', '凸', '及', '队', '乃', '中', '水', '永',
+                          '以', '山', '亡', '肅', '己', '亞', '丂', '乂', '人',
+                          '乄', '公', '巡', '火', '尺', '之', '弋', '必', '心',
+                          '狐']
+    dataset = set.union(dataset, set(basic_stroke_chars))
+
+
+    # Select one other character for each radical
+    random.seed(2066)
+
+    # Define function that add character to dataset based on radical
+    def add_char_by_radical(df, dataset, repeat=1):
+        for j in range(repeat):
+            for i in range(214):
+                chars = df.loc[df['radical_number'] == i + 1]['character']\
+                    .tolist()
+                char = random.choice(chars)
+                dataset.add(char)
+
+    # Add one character for each radical
+    add_char_by_radical(df, dataset)
+
+
+    # Define function that add character to dataset based n total number of
+    # strokes
+    def add_char_by_stroke(df, dataset, repeat=1):
+        stroke_list = df['total_strokes'].tolist()
+        stroke_unique = (set(stroke_list))
+        for j in range(repeat):
+            for stroke_num in stroke_unique:
+                chars = df.loc[df['total_strokes'] == stroke_num]['character']\
+                    .tolist()
+                char = random.choice(chars)
+                dataset.add(char)
+
+    # Add 2 characters for each storke number
+    add_char_by_stroke(df, dataset, 2)
+
+    # Select simple characters (total_strokes <= 4)
+    chars = df.loc[df['total_strokes'] <= 4]['character'].tolist()
+    chars = random.choices(chars, k=80)
+    dataset = set.union(dataset, set(chars))
+
+    # Select complicated characters (total_strokes >= 27)
+    chars = df.loc[df['total_strokes'] >= 27]['character'].tolist()
+    chars = random.choices(chars, k=80)
+    dataset = set.union(dataset, set(chars))
+
+    # Define function to randomly fill dataset up to n characters
+    def add_char_up_to(df, dataset, n=1000):
+        chars = df['character'].tolist()
+        while len(dataset) < n:
+            dataset.add(random.choice(chars))
+
+    # Fill dataset to 1000 randomly
+    add_char_up_to(df, dataset, 1000)
+
+    # Name current dataset charset_1k for later use
+    charset_1k = dataset.copy()
+
+
+    # Select charset_2000
+    add_char_by_radical(df, dataset, 2)
+    add_char_by_stroke(df, dataset, 4)
+    add_char_up_to(df, dataset, 2000)
+
+    charset_2k = dataset.copy()
+
+
+    # Select charset_4000
+    add_char_by_radical(df, dataset, 4)
+    add_char_by_stroke(df, dataset, 8)
+    add_char_up_to(df, dataset, 4000)
+
+    charset_4k = dataset.copy()
+
+
+    # Generate randset_x
+    dataset = set()
+    add_char_up_to(df, dataset, 1000)
+    randset_1k = dataset.copy()
+    add_char_up_to(df, dataset, 2000)
+    randset_2k = dataset.copy()
+    add_char_up_to(df, dataset, 4000)
+    randset_4k = dataset.copy()
+
+    # Store datasets
```

Comment:
define a function
output_file(data, filename)
and then
output_file(charset_1k, 'charset_1k')
output_file(charset_2k, 'charset_2k')
output_file(charset_4k, 'charset_4k')
output_file(randset_1k, 'randset_1k')
output_file(randset_2k, 'randset_2k')
output_file(randset_4k, 'randset_4k')
output_file(df['character'], 'full_dataset')